### PR TITLE
Use cluster profile instead of target cloud

### PR DESCRIPTION
**Pull request by bbguimaraes**
/cc @smarterclayton @stevekuznetsov
/hold

### CODE REVIEW

**Comment by bbguimaraes** [comment]
PR chain: openshift/ci-operator#173 openshift/ci-operator-prowgen#32 openshift/release#1861.

File: pkg/api/types.go [hunk 1]
```diff
@@ -302,20 +301,24 @@ type ContainerTestConfiguration struct {
...
-// Types of clusters supported by the provisioners.
 const (
-	TargetCloudAWS TargetCloud = "aws"
-	TargetCloudGCP             = "gcp"
+	ClusterProfileAWS       ClusterProfile = "aws"
+	ClusterProfileAWSAtomic                = "aws-atomic"
+	ClusterProfileAWSCentos                = "aws-centos"
+	ClusterProfileGCP                      = "gcp"
+	ClusterProfileGCPHA                    = "gcp-ha"
+	ClusterProfileGCPCRIO                  = "gcp-crio"
```
Comment (bbguimaraes) [inline pkg/api/types.go]: @smarterclayton: please review these.

File: pkg/api/config.go [hunk 2]
```diff
@@ -151,11 +151,20 @@ func validateReleaseTagConfiguration(fieldRoot string, input ReleaseTagConfigura
...
 
-func validateTargetCloud(fieldRoot string, tc TargetCloud) error {
-	if tc != TargetCloudAWS && tc != TargetCloudGCP {
-		return fmt.Errorf("%q: invalid target cloud %q", fieldRoot, tc)
-	}
-	return nil
+func validateClusterProfile(fieldRoot string, p ClusterProfile) error {
+	switch p {
+	case ClusterProfileAWSAtomic:
+		fallthrough
```
Comment (stevekuznetsov) [inline pkg/api/config.go]: amazing. didn't know this was in go syntax

Comment (stevekuznetsov) [reply pkg/api/config.go]: however, the idiom is to use commas
https://gobyexample.com/switch

**Pull request merged**